### ISSUE TITLE
Each icon import resolving to separate sprite, but only one sprite in build

### ISSUE THREAD

**Issue by derekr** [mcansh/vite-plugin-svg-sprite#28]
Things work well in dev, but when I build and run that it seems like each imported svg has a unique sprite href, but there is only one in the build output.

- the generated svg sprite has all the expected symbols
- the hash in the export/generated href doesn't seem to match what's in the logs
- the last processed icon wins
- maybe obvious, but works if you don't include hash in the sprite name (like the remix example), but ideally I can use hash for long term caching

![image](https://github.com/mcansh/vite-svg-sprite-plugin/assets/22249/a49f6ca4-e47e-41bb-b433-4ee2c1d69eea)

![image](https://github.com/mcansh/vite-svg-sprite-plugin/assets/22249/1310372e-afb3-4ecc-9c2f-1d0500af3ed2)

```ts
// app/routes/_app.tsx
import { Icon } from '~/components/Icon'
import searchIconHref from '~/images/icons/affordances/search.svg';
import homeIconHref from '~/images/icons/affordances/home.svg';
import menuIconHref from '~/images/icons/affordances/menu.svg';
import masonryIconHref from '~/images/icons/layouts/Masonry.svg';

export default function AppLayout() {
  return <Icon href={menuIconHref} />
}

// app/components/Icon.tsx
export type IconProps = {
	href: string;
};

export function Icon({ href }: IconProps) {
	return (
		<svg className="size-4" aria-hidden>
			<use href={href} />
		</svg>
	);
}

// vite.config.ts
import { sentryVitePlugin } from '@sentry/vite-plugin';
import { vitePlugin as remix } from '@remix-run/dev';
import { installGlobals } from '@remix-run/node';
import { defineConfig } from 'vite';
import type { PluginOption } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { remixDevTools } from 'remix-development-tools';
import { createSvgSpritePlugin } from '@mcansh/vite-svg-sprite-plugin';

installGlobals();

const isStorybook = process.argv[1]?.includes('storybook');

const plugins: PluginOption[] = [tsconfigPaths()];
if (!isStorybook) {
	plugins.push(
		remixDevTools(),
		remix(),
		createSvgSpritePlugin(),
		sentryVitePlugin({
			org: 'basedash',
			project: 'basedash-charts',
		}),
	);
}

export default defineConfig({
	plugins,

	build: {
		sourcemap: true,
	},
});
```

<details><summary>w/ logging on</summary>
<pre>{
  'vite-svg-sprite-plugin:transform': {
    hash: 'c568c4a3d41e6177f50af3e6a65c947c43280292a058b2dae23e6f90a628d3ba13574f26dc40fca75432548492b15ee85e227ba22153ceb116b7a879d9183921',
    ssr: true,
    icons: [
      'icon-home-642a27cf7a214d3839a267b2264d955c1b22d65b77df25b16d9d3c9afe314dd021456d7b650c562041a99acf4eb1bf6861eb1b305601dbb826bc3eaf3d761052',
      'icon-search-f13149b99ecd4fb4cad40b79ded33a9944927dc7edf5ae1359de48c55fb52f837ada9dceefe6a9243a925fa73b06724d7eb38f74cc0aef2f60f4123588dfeab4',
      'icon-menu-7f8fe44f7970e543d8f3f99bc16f79fe42f61a43daa5590ea667d30d293faad19309fd30b071845fff699818cbe5c6ebede40d89875f59b1d8c8cc0eb346b946',
      'icon-Masonry-e1806630e0e671d31259885f99c79c21f687c20a25dc7bc74ea83dc01323a8bc19efe060ac428a7069d1435220a8aa03a293bf025d17c39e9aa250d4c7da47b4'
    ],
    code: 'export default "/assets/sprite-[hash].svg#icon-home-642a27cf7a214d3839a267b2264d955c1b22d65b77df25b16d9d3c9afe314dd021456d7b650c562041a99acf4eb1bf6861eb1b305601dbb826bc3eaf3d761052";'
  }
}
{
  'vite-svg-sprite-plugin:transform': {
    hash: 'c568c4a3d41e6177f50af3e6a65c947c43280292a058b2dae23e6f90a628d3ba13574f26dc40fca75432548492b15ee85e227ba22153ceb116b7a879d9183921',
    ssr: true,
    icons: [
      'icon-home-642a27cf7a214d3839a267b2264d955c1b22d65b77df25b16d9d3c9afe314dd021456d7b650c562041a99acf4eb1bf6861eb1b305601dbb826bc3eaf3d761052',
      'icon-search-f13149b99ecd4fb4cad40b79ded33a9944927dc7edf5ae1359de48c55fb52f837ada9dceefe6a9243a925fa73b06724d7eb38f74cc0aef2f60f4123588dfeab4',
      'icon-menu-7f8fe44f7970e543d8f3f99bc16f79fe42f61a43daa5590ea667d30d293faad19309fd30b071845fff699818cbe5c6ebede40d89875f59b1d8c8cc0eb346b946',
      'icon-Masonry-e1806630e0e671d31259885f99c79c21f687c20a25dc7bc74ea83dc01323a8bc19efe060ac428a7069d1435220a8aa03a293bf025d17c39e9aa250d4c7da47b4'
    ],
    code: 'export default "/assets/sprite-[hash].svg#icon-Masonry-e1806630e0e671d31259885f99c79c21f687c20a25dc7bc74ea83dc01323a8bc19efe060ac428a7069d1435220a8aa03a293bf025d17c39e9aa250d4c7da47b4";'
  }
}
{
  'vite-svg-sprite-plugin:transform': {
    hash: 'c568c4a3d41e6177f50af3e6a65c947c43280292a058b2dae23e6f90a628d3ba13574f26dc40fca75432548492b15ee85e227ba22153ceb116b7a879d9183921',
    ssr: true,
    icons: [
      'icon-home-642a27cf7a214d3839a267b2264d955c1b22d65b77df25b16d9d3c9afe314dd021456d7b650c562041a99acf4eb1bf6861eb1b305601dbb826bc3eaf3d761052',
      'icon-search-f13149b99ecd4fb4cad40b79ded33a9944927dc7edf5ae1359de48c55fb52f837ada9dceefe6a9243a925fa73b06724d7eb38f74cc0aef2f60f4123588dfeab4',
      'icon-menu-7f8fe44f7970e543d8f3f99bc16f79fe42f61a43daa5590ea667d30d293faad19309fd30b071845fff699818cbe5c6ebede40d89875f59b1d8c8cc0eb346b946',
      'icon-Masonry-e1806630e0e671d31259885f99c79c21f687c20a25dc7bc74ea83dc01323a8bc19efe060ac428a7069d1435220a8aa03a293bf025d17c39e9aa250d4c7da47b4'
    ],
    code: 'export default "/assets/sprite-[hash].svg#icon-search-f13149b99ecd4fb4cad40b79ded33a9944927dc7edf5ae1359de48c55fb52f837ada9dceefe6a9243a925fa73b06724d7eb38f74cc0aef2f60f4123588dfeab4";'
  }
}
{
  'vite-svg-sprite-plugin:transform': {
    hash: 'c568c4a3d41e6177f50af3e6a65c947c43280292a058b2dae23e6f90a628d3ba13574f26dc40fca75432548492b15ee85e227ba22153ceb116b7a879d9183921',
    ssr: true,
    icons: [
      'icon-home-642a27cf7a214d3839a267b2264d955c1b22d65b77df25b16d9d3c9afe314dd021456d7b650c562041a99acf4eb1bf6861eb1b305601dbb826bc3eaf3d761052',
      'icon-search-f13149b99ecd4fb4cad40b79ded33a9944927dc7edf5ae1359de48c55fb52f837ada9dceefe6a9243a925fa73b06724d7eb38f74cc0aef2f60f4123588dfeab4',
      'icon-menu-7f8fe44f7970e543d8f3f99bc16f79fe42f61a43daa5590ea667d30d293faad19309fd30b071845fff699818cbe5c6ebede40d89875f59b1d8c8cc0eb346b946',
      'icon-Masonry-e1806630e0e671d31259885f99c79c21f687c20a25dc7bc74ea83dc01323a8bc19efe060ac428a7069d1435220a8aa03a293bf025d17c39e9aa250d4c7da47b4'
    ],
    code: 'export default "/assets/sprite-[hash].svg#icon-menu-7f8fe44f7970e543d8f3f99bc16f79fe42f61a43daa5590ea667d30d293faad19309fd30b071845fff699818cbe5c6ebede40d89875f59b1d8c8cc0eb346b946";'
  }
}✓ 109 modules transformed.
build/server/.vite/manifest.json                  0.56 kB
build/server/assets/loading-light-Xt0LoAmu.gif   12.39 kB
build/server/assets/tailwind-XJirB1AP.css        34.00 kB
build/server/index.js                           166.96 kB │ map: 293.31 kB
{
  WRITE_BUNDLE: {
    hash: 'c568c4a3d41e6177f50af3e6a65c947c43280292a058b2dae23e6f90a628d3ba13574f26dc40fca75432548492b15ee85e227ba22153ceb116b7a879d9183921',
    ssr: true,
    icons: [
      'icon-home-642a27cf7a214d3839a267b2264d955c1b22d65b77df25b16d9d3c9afe314dd021456d7b650c562041a99acf4eb1bf6861eb1b305601dbb826bc3eaf3d761052',
      'icon-search-f13149b99ecd4fb4cad40b79ded33a9944927dc7edf5ae1359de48c55fb52f837ada9dceefe6a9243a925fa73b06724d7eb38f74cc0aef2f60f4123588dfeab4',
      'icon-menu-7f8fe44f7970e543d8f3f99bc16f79fe42f61a43daa5590ea667d30d293faad19309fd30b071845fff699818cbe5c6ebede40d89875f59b1d8c8cc0eb346b946',
      'icon-Masonry-e1806630e0e671d31259885f99c79c21f687c20a25dc7bc74ea83dc01323a8bc19efe060ac428a7069d1435220a8aa03a293bf025d17c39e9aa250d4c7da47b4'
    ]
  }
}
✓ built in 452ms</pre>
</details> 

During Dev
![image](https://github.com/mcansh/vite-svg-sprite-plugin/assets/22249/67abf4c0-a73e-4083-92de-717e6a62d092)

**Comment by derekr** [comment]
Tried poking around the code to debug/getting things working, but for whatever reason the `id === resolvedVirtualModuleId` is never true and the block in the `load` function is never being called. Not sure where/when the output sprite `[hash]` is getting replaced.

**Comment by mcansh** [comment]
i’ve seen this happen on and off and i haven’t been able to completely track it down yet. in the interim you can pass the spriteOutputName: `sprite.svg`

**Comment by derekr** [comment]
Yeah that's what I ended up doing for now.

**Comment by mcansh** [comment]
currently working through some albeit hacky things, but production builds should now resolve to the same correct url https://github.com/mcansh/vite-svg-sprite-plugin/pull/29 - none of the build hooks run during development, so will re-add the middleware portion i had previously back in

**Comment by mcansh** [comment]
🤖 Hello there,

We just published version `0.4.0-pre.0` which involves this issue. If you'd like to take it for a test run please try it out and let us know what you think!

Thanks!
